section-4/exercises/vsnitsar-section4.js

```diff
@@ -0,0 +1,189 @@
+    const age = currentYear - userInfo.year;
+    return `Hello, my name is ${userInfo.firstName} ${userInfo.lastName}, I'm ${age}. I work in ${userInfo.company} company.`
+}
+
+const user = {
+    firstName: "Test",
+    lastName: "User",
+    year: 2000,
+    company: "Levi9"
+};
+
+console.log(getUserInfo(user));
+
+/*
+Exercise 2:
+Create a global variable counter = 0 and write a function declaration increaseCounter() that increments
+and logs counter.
+Each time when you execute increaseCounter function global counter should be increased on 1;
+*/
+let counter = 0;
+
+function increaseCounter() {
+    counter++;
+    console.log(counter);
+}
+
+/*
+Exercise 3:
+What will show console logs in the code bellow ? Think without code execution
+const message = 'Hi User';
+
+function greet() {
+    const message = 'Hello World';
+};
+console.log(message)
+*/
+
+/*
+Answer: Will be displayed 'Hi User' because it was declared globally, instead of message from
+greet function
+*/
+
+/*
+Exercise 4:
+Create a function expression joinWords that joins multiple words into a sentence and print the result.
+Use rest operator for passing arguments in function.
+*/
+
+const joinWords = (...words) => {
+    return words.join(' ');
+}
+
+console.log(joinWords('My', 'name', 'Vlad'));
+
+/*
+Exercise 5:
+What will show console logs in the code bellow ? Think without code execution
+
+const getUserInfo = function(name = 'Default', year = 2000) {
+    return 'My name is' + ' ' + name + ', I am' + ' ' + year;
+}
+console.log(getUserInfo('Test user'));
+console.log(getUserInfo('Test user two', 2025));
+*/
+
+/*
+Answer:
+The first console.log will show 'My name is Test user, I am 2000' because we only pass one parameter,
+and for the year, JavaScript will use the defined value '2000'.
+The second console.log will show 'My name is Test user two, I am 2025' because both parameters
+are passed, and the new values are used.
+ */
+
+/*
+Exercise 6:
+Write an arrow function calculatePrice where will be two arguments "price" and "tax" that returns the total
+price after tax.
+Function should calculate the total price even if "tax" argument is not passed. Hint: use default value.
+*/
+
+const calculatedPrice = (price, tax = 5) =>
+    price + (price * tax / 100);
+
+console.log(calculatedPrice(100));
+
+/*
+Exercise 7:
+What will show each console logs in the code bellow? Think without code execution
+
+let firstName = 'Test user';
+
+if(true) {
+    const userAge = 25;
+    firstName = 'Test user two';
+    console.log(firstName);  ??
+}
+console.log(userAge); ??
+*/
+
+/*
+Answer: Will be displayed error, because userAge declared inside of if and not available globally, if comment this part
+then console.log(firstName); will show 'Test user two' because value was updated inside of if (which always true)
+ */
+
+/*
+Exercise 8:
+Declare a global object user = { name: "Alice", age: 25 }. Modify its age inside a function and log the object.
+*/
+
+let newUser = {
```

Comment:
Why you decelerate this variable as **let** ? You didn't reassign  newUser objects